libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestRequest.java

```diff
@@ -482,20 +517,15 @@ public static RestRequest getRequestForSearchResultLayout(String apiVersion, Lis
 	 *
      * @param apiVersion
      * @param allOrNone
-	 * @param refIdToRequests    Sorted map of reference id to refIdToRequests
+	 * @param refIdToRequests    Linked map of reference id to refIdToRequests (will be played in order requests were added)
 	 * @return
 	 */
-	public static RestRequest getCompositeRequest(String apiVersion, boolean allOrNone, SortedMap<String, RestRequest> refIdToRequests) throws JSONException {
+	public static RestRequest getCompositeRequest(String apiVersion, boolean allOrNone, LinkedHashMap<String, RestRequest> refIdToRequests) throws JSONException {
```

Comment:
A SortedMap keeps keys sorted / a LinkedHashMap remembers the order of insertion. It's the latter we want.